src/mongocrypt-ctx-encrypt.c

```diff
@@ -1433,6 +1451,94 @@ _fle2_strip_encryptionInformation(const char *cmd_name, bson_t *cmd /* in and ou
     return ok;
 }
 
+/*
+ * Checks the "encryptedFields.strEncodeVersion" field for "create" commands for validity, and sets it to the default if
+ * it does not exist.
+ */
+static bool _fle2_fixup_encryptedFields_strEncodeVersion(const char *cmd_name,
+                                                         bson_t *cmd /* in and out */,
+                                                         const mc_EncryptedFieldConfig_t *efc,
+                                                         mongocrypt_status_t *status) {
+    BSON_ASSERT_PARAM(cmd_name);
+    BSON_ASSERT_PARAM(cmd);
+    BSON_ASSERT_PARAM(efc);
+
+    if (0 == strcmp(cmd_name, "create")) {
+        bson_iter_t ef_iter;
+        if (!bson_iter_init_find(&ef_iter, cmd, "encryptedFields")) {
+            // No encryptedFields, nothing to check or fix
+            return true;
+        }
+        bson_iter_t sev_iter;
+        if (!bson_iter_init(&sev_iter, cmd)) {
+            CLIENT_ERR("_fle2_fixup_encryptedFields_strEncodeVersion: Failed to initialize bson_iter");
+            return false;
+        }
+        if (!bson_iter_find_descendant(&sev_iter, "encryptedFields.strEncodeVersion", &sev_iter)) {
+            if (efc->str_encode_version == 0) {
+                // Unset StrEncodeVersion matches the EFC, nothing to fix.
+                return true;
+            }
+            bool ok = false;
+            // No strEncodeVersion and the EFC has a nonzero strEncodeVersion, add it.
+            bson_t fixed = BSON_INITIALIZER;
+            bson_copy_to_excluding_noinit(cmd, &fixed, "encryptedFields", NULL);
+            bson_t ef;
+            const uint8_t *data;
+            uint32_t len;
+            if (!BSON_ITER_HOLDS_DOCUMENT(&ef_iter)) {
+                CLIENT_ERR(
+                    "_fle2_fixup_encryptedFields_strEncodeVersion: Expected encryptedFields to be type obj, got: %d",
+                    bson_iter_type(&ef_iter));
+                goto fail2;
+            }
+            bson_iter_document(&ef_iter, &len, &data);
+            if (!bson_init_static(&ef, data, len)) {
+                CLIENT_ERR("_fle2_fixup_encryptedFields_strEncodeVersion: bson_init_static failed");
+                goto fail2;
+            }
+            bson_t fixed_ef;
+            bson_copy_to(&ef, &fixed_ef);
+            if (!BSON_APPEND_INT32(&fixed_ef, "strEncodeVersion", efc->str_encode_version)) {
+                CLIENT_ERR("_fle2_fixup_encryptedFields_strEncodeVersion: Failed to append strEncodeVersion");
+                goto fail1;
+            }
+            if (!BSON_APPEND_DOCUMENT(&fixed, "encryptedFields", &fixed_ef)) {
+                CLIENT_ERR("_fle2_fixup_encryptedFields_strEncodeVersion: Failed to append encryptedFields");
+                goto fail1;
+            }
```

Comment:
can't we just do `bson_append_document_begin(&fixed, "encryptedFields")` then iterate the fields of `ef` and append them to `fixed`, instead of doing one extra `bson_t`?